src/SignalR/clients/csharp/Client/test/UnitTests/LongPollingTransportTests.cs

```diff
@@ -692,4 +694,77 @@ public async Task SendsDeleteRequestWhenTransportCompleted()
             Assert.Equal(TestUri, deleteRequest.RequestUri);
         }
     }
+
+    [Fact]
+    public async Task PollRequestsContainCorrectAcceptHeader()
+    {
+        var testHttpHandler = new TestHttpMessageHandler();
+        var responseTaskCompletionSource = new TaskCompletionSource<HttpResponseMessage>();
+        var requestCount = 0;
+        var allHeadersCorrect = true;
+        var firstRequestReceived = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+        var secondRequestReceived = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+
+        testHttpHandler.OnRequest(async (request, next, cancellationToken) =>
+        {
+            if (request.Headers.Accept?.Contains(new MediaTypeWithQualityHeaderValue("*/*")) != true)
+            {
+                allHeadersCorrect = false;
+            }
+
+            requestCount++;
+
+            if (requestCount == 1)
+            {
+                firstRequestReceived.SetResult();
+            }
+            else if (requestCount == 2)
+            {
+                secondRequestReceived.SetResult();
+            }
+
+            if (requestCount >= 2)
+            {
+                if (allHeadersCorrect)
+                {
+                    responseTaskCompletionSource.TrySetResult(new HttpResponseMessage(HttpStatusCode.OK));
+                }
+                else
+                {
+                    responseTaskCompletionSource.TrySetResult(new HttpResponseMessage(HttpStatusCode.NoContent));
+                }
+            }
+
+            return await Task.FromResult(new HttpResponseMessage(HttpStatusCode.OK));
+        });
+
+        using (var httpClient = new HttpClient(testHttpHandler))
+        {
+            var loggerFactory = NullLoggerFactory.Instance;
+            var transport = new LongPollingTransport(httpClient, loggerFactory: loggerFactory);
+
+            var cts = new CancellationTokenSource();
+            var startTask = transport.StartAsync(new Uri("http://test.com"), TransferFormat.Text, cts.Token);
+
+            await firstRequestReceived.Task;
+
+            await Task.Delay(100);
```

Comment:
We try to avoid adding delays to tests because it creates flakiness. It looks like you can just use `secondRequestReceived` and get rid of `firstRequestReceived`. I don't think canceling the cts does anything?